lib/fldabnum.gd

```diff
@@ -294,25 +294,28 @@ DeclareGlobalVariable( "GaussianRationals",
 DeclareCategory( "IsGaussianRationals", IsCyclotomicCollection and IsField );
 #T better?
 
+DeclareGlobalFunction( "AbelianNumberFieldByReducedGaloisStabilizerInfo" );
 
 #############################################################################
 ##
 #V  CYCLOTOMIC_FIELDS
 ##
 ##  <ManSection>
-##  <Var Name="CYCLOTOMIC_FIELDS"/>
+##  <Func Name="CYCLOTOMIC_FIELDS" Arg='n'/>
 ##
 ##  <Description>
-##  At position <A>n</A>, the <A>n</A>-th cyclotomic field is stored.
+##  Returns the <A>n</A>-th cyclotomic field.
 ##  </Description>
 ##  </ManSection>
 ##
-DeclareGlobalVariable( "CYCLOTOMIC_FIELDS",
-    "list, CYCLOTOMIC_FIELDS[n] = CF(n) if bound" );
-
-InstallFlushableValue( CYCLOTOMIC_FIELDS,
-    [ Rationals,,, GaussianRationals ] );
-
+BindGlobal( "CYCLOTOMIC_FIELDS",
+    MemoizePosIntFunction(
+        function(xtension)
+            return AbelianNumberFieldByReducedGaloisStabilizerInfo( Rationals,
+                   xtension, [ 1 ] );
+        end,
+    [ Rationals, Rationals,, GaussianRationals ] )
+);
```

Comment:
Will you also apply these changes to `hpcgap/lib/fldabnum.gi` and `hpcgap/lib/ffe.g`? That was kind of the original reason for the new interface :-)